server/src/transport/ws.rs

```diff
@@ -246,12 +261,24 @@ async fn send_task(
 			// Handle timer intervals.
 			Either::Right((Either::Left((_instant, _stopped)), next_rx)) => {
 				stop = _stopped;
+				let now = Instant::now();
 				if let Err(err) = send_ping(&mut ws_sender).await {
 					tracing::debug!(target: LOG_TARGET, "WS send ping error: {}", err);
 					break;
 				}
 
+				if now.elapsed() > Duration::from_secs(30) {
+					tracing::warn!(target: LOG_TARGET, "Send ping was slow {}s, peer={:?}", now.elapsed().as_secs(), remote_addr);
+					break;
+				}
+
 				rx_item = next_rx;
+
+				let ping_tx = ping_tx.clone();
+				tokio::spawn(async move {
+					ping_tx.send(KeepAlive::Ping(Instant::now())).await.ok();
+				});
```

Comment:
If there is no ping_rx (ie because no ping config, so it's dropped ie on https://github.dev/paritytech/jsonrpsee/blob/705148f0c05dca1bf945e2387a866561a58beac5/server/src/transport/ws.rs#L92), I guess we are just doing this work for no gain. Is it worth putting an if around the task spawning to avoid in this case or does it over complicate things?

Reply:
good catch, I can make it an option instead.